vulnerabilities/importers/nvd.py

```diff
@@ -20,150 +20,175 @@
             )
-
-        return severity_scores
-
-    def extract_reference_urls(self, cve_item):
-        urls = set()
-        for reference in cve_item["cve"]["references"]["reference_data"]:
-            ref_url = reference["url"]
-
-            if not ref_url:
-                continue
-
-            if ref_url.startswith("http") or ref_url.startswith("ftp"):
-                urls.add(ref_url)
-
-        return urls
-
-    def is_outdated(self, cve_item):
-        cve_last_modified_date = cve_item["lastModifiedDate"]
-        cve_last_modified_date_obj = dateparser.parse(cve_last_modified_date)
-
-        if self.config.cutoff_date:
-            return cve_last_modified_date_obj < self.config.cutoff_date
-
-        if self.config.last_run_date:
-            return cve_last_modified_date_obj < self.config.last_run_date
-
-        return False
-
-    def related_to_hardware(self, cve_item):
-        for cpe in self.extract_cpes(cve_item):
-            cpe_comps = cpe.split(":")
-            # CPE follow the format cpe:cpe_version:product_type:vendor:product
-            if cpe_comps[2] == "h":
-                return True
-
-        return False
-
-    @staticmethod
-    def extract_cpes(cve_item):
-        cpes = set()
-        for node in cve_item["configurations"]["nodes"]:
-            for cpe_data in node.get("cpe_match", []):
+        )
+        if "https://nvd.nist.gov/vuln/detail/{cve_id}" in ref_urls:
+            ref_urls.remove(f"https://nvd.nist.gov/vuln/detail/{cve_id}")
+        references.extend([Reference(url=url) for url in ref_urls])
+        if cve_id:
+            aliases.append(cve_id)
+        summary = extract_summary(cve_item)
+        yield AdvisoryData(
+            aliases=aliases,
+            summary=summary,
+            references=sorted(references),
+            date_published=dateparser.parse(cve_item.get("publishedDate")),
+        )
+
+
+def extract_reference_urls(cve_item):
+    """
+    Return a list of reference URLs for a given CVE item.
+    """
+    urls = set()
+    for reference in get_item(cve_item, "cve", "references", "reference_data") or []:
+        ref_url = reference.get("url")
+
+        if not ref_url:
+            continue
+
+        if ref_url.startswith("http") or ref_url.startswith("ftp"):
+            urls.add(ref_url)
+
+    return urls
+
+
+def related_to_hardware(cpes):
+    """
+    Return True if the CVE item is related to hardware.
+    """
+    for cpe in cpes:
+        cpe_comps = cpe.split(":")
+        # CPE follow the format cpe:cpe_version:product_type:vendor:product
+        if len(cpe_comps) > 2 and cpe_comps[2] == "h":
+            return True
+
+    return False
+
+
+def extract_cpes(cve_item):
+    """
+    Return a list of CPEs for a given CVE item.
+    """
+    cpes = set()
+    for node in get_item(cve_item, "configurations", "nodes") or []:
+        for cpe_data in node.get("cpe_match") or []:
+            if cpe_data.get("cpe23Uri"):
```

Comment:
Minor nitpick to avoid getting the value of `cpe23Uri` from `cpe_data` twice.

```suggestion
            cpe23Uri_value = cpe_data.get("cpe23Uri")
            if cpe23Uri_value:
                cpes.add(cpe23Uri_value)
```